Add bookmarks menu to CyclesShaderUI.py

BrianHanke opened the pull request:
Following up on our earlier chat, I found that the Cycles image texture file picker needed a bookmark menu as well.

johnhaddon commented on python/GafferCyclesUI/CyclesShaderUI.py:
```diff
@@ -84,7 +84,12 @@ def __translateParamMetadata( nodeTypeName, socketName, value ) :
+		__metadata[paramPath]["path:leaf"] = True
+		__metadata[paramPath]["path:valid"] = True
+		__metadata[paramPath]["path:bookmarks"] = "texture"
+		__metadata[paramPath]["fileSystemPath:extensions"] = "exr tif tiff hdr jpg jpeg png"
```
When I did the equivalent for Arnold, I only included the `tx` extension, to encourage folks to use properly mip-mapped textures. Perhaps it's worth doing the same here? Or perhaps at the very least we could include `tx` in this list? I _think_ Cycles uses OIIO for image loading, so if we wanted a complete list we could get it from `GafferImageImageReader.supportedExtensions()`.

johnhaddon reviewed (commented):
Thanks Brian, and congrats on your first PR! I'll leave it up to you how to resolve my question about `tx` in the extensions list, as long as it is included one way or another.

It would be great if you could also [update Changes.md](https://github.com/gafferHQ/gaffer/wiki/Editing-Changes.md) to include a comment in the Improvements section, similar to the one we added when doing this for Arnold.

As it stands, this PR will be merged to `main` only, which means it will only appear in Gaffer 1.4. If you'd like it in 1.3 then let us know and I can rejig it appropriately.

Cheers...
John

BrianHanke commented:
Thanks @johnhaddon! I updated it to use supportedExtensions() and that seems to be working well. The browser is showing jpg, exr, tx, etc now. I also added Changes.md, hope the syntax is correct.

johnhaddon commented:
Thanks Brian! I've merged this to the `1.3_maintenance` branch, so it will appear in Gaffer 1.3.10.0. For future reference, I moved the content of your additional `python/GafferCyclesUI/Changes.md` into the main Changes.md (we just manage a single file for changes, not one per directory).

BrianHanke commented:
Great, thank you! I'll remember that for next time.
